pkg/cortex/cortex.go

```diff
@@ -190,6 +190,11 @@ func (c *Config) Validate(log log.Logger) error {
 	if err := c.TableManager.Validate(); err != nil {
 		return errors.Wrap(err, "invalid table_manager config")
 	}
+
+	if c.Storage.Engine == storage.StorageEngineBlocks && len(c.Schema.Configs) > 0 {
```

Comment:
It's slightly more tricky because we also need to check for the secondary storage engine (used to migrate between storage engines). I'm going to push a fix for that.